### ISSUE TITLE
PrimaryScrollController.of() was called with a context that does not contain a PrimaryScrollController widget error

### ISSUE THREAD

**Issue by karen1au** [optimist-dev/scrolls_to_top#7]
I updated to flutter 3.7.0 and start getting the following error on build:
```
FlutterError (PrimaryScrollController.of() was called with a context that does not contain a PrimaryScrollController widget.
No PrimaryScrollController widget ancestor could be found starting from the context that was passed to PrimaryScrollController.of(). This can happen because you are using a widget that looks for a PrimaryScrollController ancestor, but no such ancestor exists.
The context used was:
  Navigator-[LabeledGlobalKey<NavigatorState>#59c48](dependencies: [HeroControllerScope, MediaQuery, UnmanagedRestorationScope, _FocusTraversalGroupMarker], state: NavigatorState#82df9(tickers: tracking 1 ticker)))
```
The error stack trace point to here
<img width="980" alt="image" src="https://user-images.githubusercontent.com/43124859/214649700-ea160adf-4f7d-4117-9c7c-1b5003b4fd43.png">

I'm using Scaffold as the child of the `ScrollsToTop` widget, it was working before 🤔 
I tried declaring the controller as:
```
final primaryScrollController = PrimaryScrollController.of(context);
```
and the error is gone and scroll to top also seem to be working, wondering if there's any specific use case where using `PrimaryScrollController.of(Navigator.of(context).context)` is necessary?

Thanks in advance!

**Comment by dkbast** [comment]
I just switched back to 1.1.0 for now, and that seems to be working with Flutter 3.7 in my case, the change @karen1au tries is actually the only difference between 2.0.0 and 1.1.0, so you don't have to fork it to try this for yourself

**Comment by otopba** [comment]
Fixed at version 2.1.0. Thanks!
